srv/src/main/java/my/bookshop/handlers/HierarchyHandler.java

```diff
@@ -0,0 +1,304 @@
+package my.bookshop.handlers;
+
+import java.util.ArrayDeque;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import com.sap.cds.Row;
+import com.sap.cds.ql.CQL;
+import com.sap.cds.ql.Select;
+import com.sap.cds.ql.cqn.CqnElementRef;
+import com.sap.cds.ql.cqn.CqnPredicate;
+import com.sap.cds.ql.cqn.CqnSelect;
+import com.sap.cds.ql.cqn.CqnValue;
+import com.sap.cds.ql.cqn.Modifier;
+import com.sap.cds.ql.cqn.transformation.CqnTopLevelsTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnAncestorsTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnDescendantsTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnFilterTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnSearchTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnOrderByTransformation;
+import com.sap.cds.ql.cqn.transformation.CqnTransformation;
+import com.sap.cds.services.cds.CdsReadEventContext;
+import com.sap.cds.services.cds.CqnService;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
+
+import cds.gen.adminservice.AdminService_;
+import cds.gen.adminservice.GenreHierarchy;
+import cds.gen.adminservice.GenreHierarchy_;
+
+import static cds.gen.adminservice.AdminService_.GENRE_HIERARCHY;
+
```

Comment:
please add javadoc which explains that this handler is only a stand-in as long as we can't handler hierarchy transformations generically on non-HANA (H2)

please also add a warning that this handler is not necessarily functionally correct and should not be used as a blue-print